src/prefect/tasks/docker/images.py

```diff
@@ -119,8 +120,27 @@ def run(
             raise ValueError("A repository to pull the image from must be specified.")
 
         client = docker.APIClient(base_url=docker_server_url, version="auto")
+        self.logger.debug(
+            "Starting docker pull for repository {repo} with tag {tag}...".format(
+                repo=repository, tag=tag
+            )
+        )
+        feed = client.pull(repository=repository, tag=tag, stream=True)
+
+        output = ""
+        return_stream, log_stream = itertools.tee(feed)
+
```

Comment:
Planning on removing the itertools.tee call, as I remember a slack chat that Prefect isn't "not not threadsafe", so just going to append to the output string.

Reply:
Actually not fully sure about this one but simply appending to the output string wouldn't hurt.

cc @cicdw